server/grpc_service.go

```diff
@@ -513,13 +506,92 @@ func (s *GrpcServer) forwardTSO(stream pdpb.PD_TsoServer) error {
 	}
 }
 
+func (s *GrpcServer) forwardTSORequestWithDeadLine(
+	ctx context.Context, request *pdpb.TsoRequest, forwardStream tsopb.TSO_TsoClient,
+) (*tsopb.TsoResponse, error) {
+	// Create a context with deadline for forwarding TSO request to TSO service.
+	ctxTimeout, cancel := context.WithTimeout(ctx, tsoutil.DefaultTSOProxyTimeout)
+	defer cancel()
+
+	// used to receive the result from doSomething function
+	tsoRespCh := make(chan *tsopbTSOResponse)
```

Comment:
Should we use the block channel?

Reply:
You're right, we should use tsoRespCh := make(chan *tsopbTSOResponse, 1)

Reply:
changed